content/general/events.textile

```diff
@@ -0,0 +1,374 @@
+- Channel filter := An optional filter, which allows the Rule to be applied to a restricted set of channels. This can be specified as a regular expression, allowing for swathes of channels to be used
+- "Encoding":#encoding := The encoding to be used by this Rule. This can be either JSON or "MsgPack":http://msgpack.org
+
+If the Rule is in the *Single Request* mode, it will also have the following options:
+
+- "Enveloped":#envelope := If the rule has the Enveloped option set, then data delivered by this Rule will be wrapped in an "Ably envelope":#envelope. Otherwise, the Rule will send the "raw payload":#no-envelope-examples
+
+If the Rule is in the *Batch Request* mode, it will have the following additional options:
+
+- Sign with key := Ably will optionally sign the data with the specified private key. This will be included as an HTTP header @X-Ably-Signature@ in every HTTP post request issued to your server. See "WebHook security":#security for more details.
+
+*Note* that various integrations have restrictions on them which will mean some of these base options are either changed or absent. You can check specific details in each "integration's page":#integrations.
+
+h3(#sources). Sources
+
+Ably currently supports three types of Events:
+
+* "Messages":#messages - messages trigger function calls as soon as they are published on a channel
+* "Presence events":#presence - presence events trigger function calls when clients enter, update their data, or leave channels
+* "Channel lifecycle events":/general/webhooks#channel-lifecycle – get notified when a channel is created (following the first client attaching to this channel) or discarded (when there are no more clients attached to the channel). Currently this is *only* available for "batched":#batching messages
+
+h3(#batching). Single vs Batched requests
+
+If *Single request* is chosen for a Reactor Event Rule, then a @POST@ request will be sent to your specified endpoint/service each time an event occurs. Although this can be useful for some use-cases where the endpoint can only process one message per request, or needs the event as soon as it's available, it can result in the endpoint being overloaded with requests. To avoid this, it's possible to instead make use of *Batch request* instead, which will batch messages sent within a set timeframe together.
+
+h4(#single-request). Single request details
+
+Single request is best suited for scenarios where you're wanting a 1-to-1 relationship between sent messages and events being called. If you are making use of a serverless system which is expecting a single piece of data each time, and then intends to perform some transformation/event following that, then Single request will likely work well for you. If you're using a single server, which has the potential to be overloaded by requests, and can process multiple events per payload sent, Batch request will be a better choice.
+
+minimize. View single request transport details
+  It just works
```

Comment:
- rate limits: https://support.ably.io/support/solutions/articles/3000053845-do-you-have-any-connection-message-or-rate-limits-on-accounts- under 'reactor events'. 15 free, 30 paid. (Come to think of that doesn't make any sense, we really need to rethink that.)
- request timeout: 15s
- multiple in-flight at once, up to the max concurrency (listed on the limits article). If the number of in-flight requests exceeds the max concurrency, new messages coming in are placed in a short queue (length 10); if that queue length is exceeded, further messages are rejected.
- if the post request is rejected with 5xx or times out, we retry twice more, once after 4s, then if that fails, again after 20s.